src/Microsoft.IdentityModel.Logging/ISafeLogSecurityArtifact.cs

```diff
@@ -0,0 +1,18 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+namespace Microsoft.IdentityModel.Logging
+{
+    /// <summary>
+    /// Interface that provides an unsafe method to log a security artifact.
+    /// </summary>
+    public interface ISafeLogSecurityArtifact
+    {
+        /// <summary>
+        /// Returns a string that represents the complete security artifact.
+        /// This may include sensitive information and should only be used for logging purposes.
```

Comment:
>logging [](http://example.com/codeflow?start=79&length=7)

We should not encourage the use of this for logging.

This should be used for exceptional debugging cases as it will reveal security tokens in plain text.

Reply:
@ciaozhang

Reply:
Still waiting for this change @ciaozhang